shell/renderer/electron_renderer_client.cc

```diff
@@ -34,15 +34,27 @@ bool IsDevToolsExtension(content::RenderFrame* render_frame) {
 
 }  // namespace
 
+// static
+ElectronRendererClient* ElectronRendererClient::self_ = nullptr;
+
 ElectronRendererClient::ElectronRendererClient()
     : node_bindings_(
           NodeBindings::Create(NodeBindings::BrowserEnvironment::RENDERER)),
-      electron_bindings_(new ElectronBindings(node_bindings_->uv_loop())) {}
+      electron_bindings_(new ElectronBindings(node_bindings_->uv_loop())) {
+  DCHECK(!self_) << "Cannot have two ElectronRendererClient";
+  self_ = this;
+}
 
 ElectronRendererClient::~ElectronRendererClient() {
   asar::ClearArchives();
```

Comment:
`self_ = nullptr`

Reply:
I could add this, but it didn't seem necessary given it wasn't added for ElectronBrowserMainParts https://github.com/electron/electron/blob/01dc50d8295690058fbde34b540e3b44b486bdbc/shell/browser/electron_browser_main_parts.cc#L205